stellar-multitenant/pkg/internal/provisioning/manager.go

```diff
@@ -3,26 +3,30 @@ package provisioning
 import (
 	"context"
 	"embed"
+	"errors"
 	"fmt"
 
 	"github.com/lib/pq"
 	migrate "github.com/rubenv/sql-migrate"
 	"github.com/stellar/go/support/log"
+
 	"github.com/stellar/stellar-disbursement-platform-backend/db"
 	authmigrations "github.com/stellar/stellar-disbursement-platform-backend/db/migrations/auth-migrations"
 	sdpmigrations "github.com/stellar/stellar-disbursement-platform-backend/db/migrations/sdp-migrations"
 	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
 	"github.com/stellar/stellar-disbursement-platform-backend/internal/message"
 	"github.com/stellar/stellar-disbursement-platform-backend/internal/services"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/engine/signing"
 	"github.com/stellar/stellar-disbursement-platform-backend/internal/utils"
 	"github.com/stellar/stellar-disbursement-platform-backend/stellar-auth/pkg/auth"
 	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
 )
 
 type Manager struct {
-	tenantManager   *tenant.Manager
-	db              db.DBConnectionPool
-	messengerClient message.MessengerClient
+	tenantManager    *tenant.Manager
+	db               db.DBConnectionPool
+	messengerClient  message.MessengerClient
+	signatureService signing.SignatureService
```

Comment:
What do you think about using just the DistAccountSigner in this object? You don't need the whole signatureService, do you?

The less we expose any signature-related stuff, the better, so reducing the number of signers here is preferred.

Reply:
I agree we shouldn't expose other signer functionalities if they're not needed, initializing the sig svc also requires a whole bunch of other fields we don't need when provisioning tenants too.